database/database.go

```diff
@@ -485,9 +485,12 @@ func (t *Txn) pageRange(page uint,
 // * t.access == ControlAccess
 // * channel != nil
 //
+// PutChannel preamble:
+//  oldCount := t.mustCountCh()
+//
 // PutChannel ensures:
 // * !dbc.InTest || err != nil || t.mustGetCh(channel.Descriptor_) != nil
-// * !dbc.InTest || err != nil || t.mustCountCh() > 0
+// * !dbc.InTest || err != nil || t.mustCountCh() == oldCount+1 || t.mustCountCh() == oldCount
```

Comment:
You could actually introduce oldHas into preamble's `if dbc.InTest { ... }` to make the contract a bit more precise:

```go
var hasBefore bool
if dbc.InTest {
    ...
    oldHas = t.mustGet(...)
}
```

Then the condition becomes:
```
!dbc.InTest || err != nil || (!oldHas || t.mustCountCh() == oldCount)
!dbc.InTest || err != nil || (oldHas || t.mustCountCh() == oldCount + 1)
```